backends/cortex_m/ops/cortex_m_ops_common.h

```diff
@@ -0,0 +1,184 @@
+  int64_t out_mult_val = output_multiplier.to<int64_t>();
+  int64_t out_shift_val = output_shift.to<int64_t>();
+
+  ET_CHECK_MSG(
+      zp1_val >= std::numeric_limits<int8_t>::min() &&
+          zp1_val <= std::numeric_limits<int8_t>::max(),
+      "Zero point 1 must be in int8 range [Value: %d]",
+      zp1_val);
+
+  ET_CHECK_MSG(
+      zp2_val >= std::numeric_limits<int8_t>::min() &&
+          zp2_val <= std::numeric_limits<int8_t>::max(),
+      "Zero point 2 must be in int8 range [Value: %d]",
+      zp2_val);
+
+  ET_CHECK_MSG(
+      out_zp_val >= std::numeric_limits<int8_t>::min() &&
+          out_zp_val <= std::numeric_limits<int8_t>::max(),
+      "Output zero point must be in int8 range [Value: %d]",
+      out_zp_val);
+
+  // Check multipliers fit in int32 range (AOT quantize_multiplier_aot clamps to
+  // int32)
+  ET_CHECK_MSG(
+      mult1_val >= std::numeric_limits<int32_t>::min() &&
+          mult1_val <= std::numeric_limits<int32_t>::max(),
+      "Multiplier 1 must be in int32 range [Value: %d]",
+      mult1_val);
+
+  ET_CHECK_MSG(
+      mult2_val >= std::numeric_limits<int32_t>::min() &&
+          mult2_val <= std::numeric_limits<int32_t>::max(),
+      "Multiplier 2 must be in int32 range [Value: %d]",
+      mult2_val);
+
+  ET_CHECK_MSG(
+      out_mult_val >= std::numeric_limits<int32_t>::min() &&
+          out_mult_val <= std::numeric_limits<int32_t>::max(),
+      "Output multiplier must be in int32 range [Value: %d]",
+      out_mult_val);
+
+  ET_CHECK_MSG(
+      shift1_val >= -31 && shift1_val <= 31,
+      "Shift 1 must be in range [-31, 31] [Value: %d]",
+      shift1_val);
+
+  ET_CHECK_MSG(
+      shift2_val >= -31 && shift2_val <= 31,
+      "Shift 2 must be in range [-31, 31] [Value: %d]",
+      shift2_val);
+
+  ET_CHECK_MSG(
+      out_shift_val >= -31 && out_shift_val <= 31,
+      "Output shift must be in range [-31, 31] [Value: %d]",
+      out_shift_val);
+}
+
+inline Error resize_to_broadcast_target_size_quantized(
+    const Tensor& input1,
+    const Tensor& input2,
+    Tensor& output) {
+  static constexpr int kTensorDimensionLimit = 5;
+
+  int inp1_shape[kTensorDimensionLimit];
+  int inp2_shape[kTensorDimensionLimit];
+  int out_shape[kTensorDimensionLimit];
+
+  int max_dim = std::max({input1.dim(), input2.dim(), output.dim()});
+  max_dim = std::min(max_dim, kTensorDimensionLimit);
+
+  // Initialize shapes with 1s for padding
+  for (int i = 0; i < max_dim; i++) {
+    inp1_shape[i] = 1;
+    inp2_shape[i] = 1;
+    out_shape[i] = 1;
+  }
+
+  int offset_inp1 = max_dim - input1.dim();
+  int offset_inp2 = max_dim - input2.dim();
+  int offset_out = max_dim - output.dim();
+
+  for (int i = 0; i < input1.dim(); i++) {
+    inp1_shape[i + offset_inp1] = input1.size(i);
+  }
+  for (int i = 0; i < input2.dim(); i++) {
+    inp2_shape[i + offset_inp2] = input2.size(i);
+  }
+  for (int i = 0; i < output.dim(); i++) {
+    out_shape[i + offset_out] = output.size(i);
+  }
```

Comment:
do we need all this logic if we are going to call the `get_broadcast_target_size` util?